### PR TITLE
[GetChainChangedUseCase] create use case which returns stream when chain is changed on MetaMask

### PR DESCRIPTION
**Pull request by mretana1999**
# Description
We need a use case that will provide a stream to listen for chain changes so that all the UI elements are notified of these events.

Use ethereum object from package:flutter_web3/flutter_web3.dart which contains a onChainChanged method that we can subscribe to listen for changes on chain

Provide a new stream that emits which chain the user is connected to or if the user is connected to an unsupported chain.

Fixes Jira Ticket # 597

# Additional Notes: 
We are unable to create a unit test for this use case at this moment. This is because tests are run in Dart VM, and not in a browser that has the MetaMask extension, so some packages are not available when test is run.

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Is this a UI Change? If so please include screenshot of before and after states below

# How Has This Been Tested?
Application was run with Chrome in web-server mode:
image

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have assigned 2 reviewers to check my work
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/service/Controller/usecases/GetChainChangesUseCase_TODOtest.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  setUp(() {
+    mockEthereumChainWrapper = MockEthereumWebChainWrapper();
+    getChainChangesUseCase = GetChainChangesUseCase(mockEthereumChainWrapper);
+    activeChainStream = getChainChangesUseCase.stream;
+  });
+
+  // test('should return a stream whenever we call the getter method', () {
+  //   when(getChainChangesUseCase.stream);
+  // });
```
Comment (Friendlygeckos) [inline test/service/Controller/usecases/GetChainChangesUseCase_TODOtest.dart:25]: Remove this if we are not using it

**Review by Friendlygeckos — CHANGES_REQUESTED** [review]

File: lib/pages/scout/bloc/ScoutPageBloc.dart [hunk 2]
```diff
@@ -1,37 +1,56 @@
...
 import 'package:ax_dapp/util/BlocStatus.dart';
 import 'package:ax_dapp/util/SupportedSports.dart';
+import 'package:ax_dapp/util/CurrentChain.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 import '../models/ScoutPageEvent.dart';
 
 class ScoutPageBloc extends Bloc<ScoutPageEvent, ScoutPageState> {
   final GetScoutAthletesDataUseCase repo;
+  final List<AthleteScoutModel> mlbAthletes = [];
```
Comment (Friendlygeckos) [inline lib/pages/scout/bloc/ScoutPageBloc.dart:14]: Remove it if we are not using it.

**Review by Friendlygeckos — CHANGES_REQUESTED** [review]

**Review by pearlson16 — APPROVED** [review]

File: lib/pages/scout/bloc/ScoutPageBloc.dart [hunk 3]
```diff
@@ -1,37 +1,55 @@
...
+  final GetChainChangesUseCase chainChangeUseCase;
+  late Stream _chainChangeStream;
 
-  ScoutPageBloc({required this.repo}) : super(ScoutPageState.initial()) {
+  ScoutPageBloc({required this.repo, required this.chainChangeUseCase})
+      : super(ScoutPageState.initial()) {
     on<OnPageRefresh>(_mapRefreshEventToState);
     on<SelectSport>(_mapSelectSportToState);
     on<OnAthleteSearch>(_mapSearchAthleteEventToState);
+    _chainChangeStream = chainChangeUseCase.stream;
```
Comment (ketonkss4) [inline lib/pages/scout/bloc/ScoutPageBloc.dart:22]: this is unnecessary

File: lib/pages/scout/bloc/ScoutPageBloc.dart [hunk 4]
```diff
@@ -1,37 +1,55 @@
...
   }
 
   void _mapRefreshEventToState(
       OnPageRefresh event, Emitter<ScoutPageState> emit) async {
     try {
       emit(state.copyWith(status: BlocStatus.loading));
-
+      _chainChangeStream.listen((event) {
```
Comment (ketonkss4) [inline lib/pages/scout/bloc/ScoutPageBloc.dart:29]: This listening event should be done on initialization of the block. Doing it this way means you are setting a new listener reference everytime you have a refresh event

Instead set up your bloc to subscribe to the stream events this way

**Review by ketonkss4 — CHANGES_REQUESTED** [review]
It looks okay for the most part but the way you have set up the bloc is awkward

File: lib/pages/scout/bloc/ScoutPageBloc.dart (hunk 4 above)
Comment (mretana1999) [reply lib/pages/scout/bloc/ScoutPageBloc.dart:29]: Listener is now moved into bloc initialization

File: lib/pages/scout/bloc/ScoutPageBloc.dart (hunk 3 above)
Comment (mretana1999) [reply lib/pages/scout/bloc/ScoutPageBloc.dart:22]: This references the getter method for obtaining broadcast stream inside usecase class.

**Review by Friendlygeckos — APPROVED** [review]

**Review by narcodico — CHANGES_REQUESTED** [review]

**Pull request closed without merging**